build/php.m4

```diff
@@ -2703,3 +2703,58 @@ AC_DEFUN([PHP_PATCH_CONFIG_HEADERS], [
   $SED -e 's/^#undef PACKAGE_[^ ]*/\/\* & \*\//g' < $srcdir/$1 \
     > $srcdir/$1.tmp && mv $srcdir/$1.tmp $srcdir/$1
 ])
+
+dnl
+dnl PHP_CHECK_AVX512_SUPPORTS
+dnl
+AC_DEFUN([PHP_CHECK_AVX512_SUPPORTS], [
+  AC_MSG_CHECKING([for avx512 supports in compiler])
+  save_CFLAGS="$CFLAGS"
+  CFLAGS="-mavx512f -mavx512cd -mavx512vl -mavx512dq -mavx512bw $CFLAGS"
+
+  AC_LINK_IFELSE([AC_LANG_SOURCE([[
+    #include <immintrin.h>
+      int main() {
```

Comment:
It is recommended to use 

# include <intrin.h>

with VC

Reply:
.mp4 files aren't used with MSVC builds, though.

Reply:
the simd parts look good to me. I did not run it but it matches the base64 implementation from a review point of view.

Also the base is there to add avx256 or sse4. The sooner is well spread these days. AVX512 mostly on the devs machines only:)